cli/sawtooth_cli/config.py

```diff
@@ -75,41 +106,87 @@ def _do_config_set(args):
     """
     settings = [s.split('=', 1) for s in args.setting]
 
-    with open(args.key, 'r') as key_file:
-        wif_key = key_file.read().strip()
-        signing_key = signing.encode_privkey(
-            signing.decode_privkey(wif_key, 'wif'), 'hex')
-        pubkey = signing.encode_pubkey(
-            signing.generate_pubkey(signing_key), 'hex')
+    pubkey, signing_key = _read_signing_keys(args.key)
 
-    txns = [_create_config_txn(pubkey, signing_key, setting)
+    txns = [_create_propose_txn(pubkey, signing_key, setting)
             for setting in settings]
-    txn_ids = [txn.header_signature for txn in txns]
 
-    batch_header = BatchHeader(signer_pubkey=pubkey,
-                               transaction_ids=txn_ids).SerializeToString()
-
-    batch = Batch(
-        header=batch_header,
-        header_signature=signing.sign(batch_header, signing_key),
-        transactions=txns
-    )
+    batch = _create_batch(pubkey, signing_key, txns)
 
     batch_list = BatchList(batches=[batch]).SerializeToString()
 
     try:
         with open(args.output, 'wb') as batch_file:
             batch_file.write(batch_list)
-    except:
-        raise CliException('Unable to write to {}'.format(args.output))
+    except IOError as ioe:
+        raise CliException(
+            'Unable to write to {}: {}'.format(args.output, ioe))
```

Comment:
IOError will often provide the filename in it's message, so I'd expect the above to contain the name of the file twice.  This can be tested by setting the permissions on the directory such that you can't write to it.